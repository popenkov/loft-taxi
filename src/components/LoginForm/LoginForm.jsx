import React, { useState } from 'react';
import { PropTypes } from 'prop-types';

import { Link, useNavigate } from 'react-router-dom';

import { ROUTES } from '../../constants/constants';
import styles from './LoginForm.module.scss';
// import { Input } from './Input';
import { Button } from '../UI/Button';
import { Input } from '../UI/FormElements/Input';

export const LoginForm = ({ title, handleChangePage }) => {
  const [formState, setFormState] = useState({
    email: '',
    password: '',
  });

  const navigate = useNavigate();

  const handleSubmitClick = (evt) => {
    evt.preventDefault();
    // TODO  navigate(ROUTES.MAP);
    handleChangePage('map');
  };

  const handleRegisterClick = () => {
    handleChangePage('register');
  };

  const handleInputChange = (evt) => {
    const inputValue = evt.target.value;
    const inputName = evt.target.name;
    setFormState({ ...formState, [inputName]: inputValue });
  };
  return (
    <div className={styles.form}>
      <h2 className={styles.title}>{title}</h2>
      <form>
        <Input
          label={'Email:'}
          id="email"
          value={formState.login}
          type="email"
          name="email"
          changeHandler={handleInputChange}
        />
        <Input
          label={'Пароль:'}
          id="password"
          value={formState.password}
          type="password"
          name="password"
          changeHandler={handleInputChange}
        />

        <Link to="/" className={styles.forgotPassword}>
          Забыли пароль?
        </Link>

        <Button text="Войти" type="primary" clickHandler={handleSubmitClick} />
      </form>
      <div className={styles.switchFormContainer}>
        <span className={styles.switchFormText}> Уже зарегистрированы?</span>
        <Button
          text="Регистрация"
          type="secondary"
          clickHandler={handleRegisterClick}
        />
      </div>
    </div>
  );
};

LoginForm.propTypes = {
  title: PropTypes.string.isRequired,
  handleChangePage: PropTypes.func,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};
