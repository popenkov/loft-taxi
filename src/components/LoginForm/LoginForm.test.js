import { screen, fireEvent } from '@testing-library/react';
import { renderTestApp } from '../../test/helpers/renderTestApp';

import { LoginForm } from './LoginForm';

describe('login form component', () => {
  let container;

  beforeEach(() => {
    container = renderTestApp(<LoginForm />, {
      route: '/',
      initialState: {},
    });
  });

  test('renders correctly', async () => {
    expect(<LoginForm />).toMatchSnapshot();
  });
});
