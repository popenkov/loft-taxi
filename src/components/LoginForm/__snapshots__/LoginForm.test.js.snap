// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login form component renders correctly 1`] = `<LoginForm />`;
